tests/unit/backend/service/test_MetadataLookupService.py

```diff
@@ -0,0 +1,27 @@
+import unittest
+
+import pytest
+from snapred.backend.service.MetadataLookupService import MetadataLookupService
+
+
+class TestMetadataLookupService(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.instance = MetadataLookupService()
+        super().setUpClass()
+
+    def setUp(self):
+        return super().setUp()
+
+    def test_verifyMultipleRuns(self):
+        # Test valid number of runs
+        runs = self.instance.verifyMultipleRuns("46342:46352")
+        assert len(runs) == 10
+
+        # Test exceeding max number of runs
+        runs = self.instance.verifyMultipleRuns("46342:46355")
+        assert len(runs) == 0
```

Comment:
You can capture and verify the warning message with [`assertLogs`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs)

Reply:
Near the top you need to import the service module specifically with
``` python
ServiceModule = importlib.import_module(MetadataLookupService.__module__)
```
then you can verify the log with
``` python
        with self.assertLogs(logger=ServiceModule.logger, level=logging.WARNING) as cm:
            runs = self.instance.verifyMultipleRuns("46342:46355")
        assert f"whatever the message is" in cm.output[0]
```